### PR TITLE
local to external db migration guide

### PR DESCRIPTION
**Pull request by gazarenkov**
## Description

Added local to external db migration guide and helper script

## Which issue(s) does this PR fix or relate to

https://issues.redhat.com/browse/RHIDP-2385

## PR acceptance criteria

- [X] Documentation

## How to test changes / Special notes to the reviewer

Try to migrate from the local to external DB according to instruction

### CODE REVIEW

File: docs/db_merge.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+backstage_plugin_app        | postgres | UTF8     | libc            | en_US.UTF-8 | en_US.UTF-8 | 
+backstage_plugin_auth       | postgres | UTF8     | libc            | en_US.UTF-8 | en_US.UTF-8 |
+backstage_plugin_catalog    | postgres | UTF8     | libc            | en_US.UTF-8 | en_US.UTF-8 |
+backstage_plugin_permission | postgres | UTF8     | libc            | en_US.UTF-8 | en_US.UTF-8 |
+backstage_plugin_scaffolder | postgres | UTF8     | libc            | en_US.UTF-8 | en_US.UTF-8 |
+backstage_plugin_search     | postgres | UTF8     | libc            | en_US.UTF-8 | en_US.UTF-8 |
+postgres                    | postgres | UTF8     | libc            | en_US.UTF-8 | en_US.UTF-8 |
+````
+
+To move the data from working Backstage instance hosted on a local PostgreSQL server to a production-ready PostgreSQL service (such as AWS RDB or Azure Database), you can use directly PostgreSQL utilities such as pg-dump with pgsql or pgAdmin and move the data from each database one-by-one.
```
Comment (rm3l) [inline docs/db_merge.md:19]: Links to such tools might be helpful.

```suggestion
To move the data from working Backstage instance hosted on a local PostgreSQL server to a production-ready PostgreSQL service (such as AWS RDB or Azure Database), you can use directly PostgreSQL utilities such as [pg_dump](https://www.postgresql.org/docs/current/app-pgdump.html) with [psql](https://www.postgresql.org/docs/current/app-psql.html) or [pgAdmin](https://www.pgadmin.org/) and move the data from each database one-by-one.
```

File: hack/db_copy.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+#TO_PSW=postgres
+to_host=rhdb.cx6ygmy4go45.eu-north-1.rds.amazonaws.com
```
Comment (rm3l) [inline hack/db_copy.sh:3]: Why is this hard-coded? This should be an input argument, no? Same comments about the other `{from,to}_*` parameters. Or if it is provided as an example, maybe you could rename it accordingly, like: `db_copy.example.sh`.

File: docs/db_merge.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+To simplify this process, we have a [**db_copy.sh**](../hack/db_copy.sh) script.
+
+### Prerequisites
+
+- [**pg_dump**](https://www.postgresql.org/docs/current/backup-dump.html) and [**pgsql**](https://www.postgresql.org/docs/current/app-psql.html) client utilities installed on your local machine.
```
Comment (rm3l) [inline docs/db_merge.md:24]:
```suggestion
- [**pg_dump**](https://www.postgresql.org/docs/current/backup-dump.html) and [**psql**](https://www.postgresql.org/docs/current/app-psql.html) client utilities installed on your local machine.
```

File: docs/db_merge.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+/opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1793                                                                                                                                  │
+          throw new errors.ForwardedError(                                                                                                                                                                        │
+                ^                                                                                                                                                                                                 │
+ MigrationLocked: Plugin 'auth' startup failed; caused by MigrationLocked: Migration table is already locked                                                                                                       │
+````                       
+
+A way to make it work w/o knex utility is to delete the data from the **knex_migrations_lock** table for each problematic Backstage plugin's database (in the example above it is **'auth'** plugin, so corresponding database is **backstage_plugin_auth**):
```
Comment (rm3l) [inline docs/db_merge.md:124]:
```suggestion
A way to make it work without knex utility is to delete the data from the **knex_migrations_lock** table for each problematic Backstage plugin's database (in the example above it is **'auth'** plugin, so corresponding database is **backstage_plugin_auth**):
```

File: docs/db_merge.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+````
+psql -h <to_host> -U <to_user> -d <database> -c "delete from knex_migrations_lock;"
+````
+
+**NOTE:** in some DB the table called differently like: **backstage_backend_tasks__knex_migrations_lock** in the **backstage_plugin_search** database
```
Comment (rm3l) [inline docs/db_merge.md:130]:
```suggestion
**NOTE:** in some DB the table is called differently like: **backstage_backend_tasks__knex_migrations_lock** in the **backstage_plugin_search** database
```

File: hack/db_copy.sh (hunk 2 above)
Comment (gazarenkov) [reply hack/db_copy.sh:3]: well, it is stated that one have to create copy of this file and provide values here.
I am not sure arguments are more convenient than modifying fields here especially that many of the fields most probably will be unchanged.
So, I would "recommend" password only  for argument, for security reason.
WDYT?

Comment (rm3l) [reply hack/db_copy.sh]: Alright. Fine then if it is stated that a copy of the file should be created.

**Review by rm3l — APPROVED** [review]

**Pull request merged**